apps/prairielearn/src/lib/server-jobs.ts

```diff
@@ -67,6 +67,19 @@ const JobSequenceWithJobsSchema = JobSequenceSchema.extend({
     }),
   ),
 });
+export type JobSequenceWithJobs = z.infer<typeof JobSequenceWithJobsSchema>;
+
+export type JobSequenceWithTokens = Omit<JobSequenceWithJobs, 'jobs'> & {
+  token: string;
+  jobs: (JobSequenceWithJobs['jobs'][0] & { token: string })[];
```

Comment:
`JobSequenceWithJobs['jobs'][0]` is fine, though you might also consider declaring a separate schema/type for that. We'd probably call that `JobRow` or similar.